arm/Microsoft.Network/firewallPolicies/deploy.bicep

```
@@ -0,0 +1,189 @@
+param fqdns array = []
+
+@description('Optional. List of IP addresses for the ThreatIntel Allowlist.')
+param ipAddresses array = []
+
+@description('Optional. Secret Id of (base-64 encoded unencrypted pfx) Secret or Certificate object stored in KeyVault.	')
+param keyVaultSecretId string = ''
+
+@description('Optional. Name of the CA certificate.')
+param certificateName string = ''
+
+@description('Optional. Customer Usage Attribution ID (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+@description('Optional. Rule collection groups.')
+param ruleCollectionGroups array = []
+
+@description('Optional. Rule groups.')
+param ruleGroups array = []
+
+var identityType = systemAssignedIdentity ? (!empty(userAssignedIdentities) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(userAssignedIdentities) ? 'UserAssigned' : 'None')
+
+var identity = identityType != 'None' ? {
+  type: identityType
+  userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
+} : null
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource firewallPolicy 'Microsoft.Network/firewallPolicies@2021-03-01' = {
+  name: name
+  location: location
+  tags: tags
+  identity: identity
+  properties: {
+    basePolicy: !empty(basePolicyResourceId) ? {
+      id: basePolicyResourceId
+    } : null
+    dnsSettings: enableProxy ? {
+      enableProxy: enableProxy
+      requireProxyForNetworkRules: requireProxyForNetworkRules
+      servers: servers
+    } : null
+    insights: isEnabled ? {
+      isEnabled: isEnabled
+      logAnalyticsResources: {
+        defaultWorkspaceId: {
+          id: !empty(defaultWorkspaceId) ? defaultWorkspaceId : null
+        }
+        workspaces: !empty(workspaces) ? workspaces : null
+      }
+      retentionDays: retentionDays
+    } : null
+    intrusionDetection: (mode != 'Off') ? {
+      configuration: {
+        bypassTrafficSettings: !empty(bypassTrafficSettings) ? bypassTrafficSettings : null
+        signatureOverrides: !empty(signatureOverrides) ? signatureOverrides : null
+      }
+      mode: mode
+    } : null
+    sku: {
+      tier: tier
+    }
+    snat: !empty(privateRanges) ? {
+      privateRanges: privateRanges
+    } : null
+    threatIntelMode: threatIntelMode
+    threatIntelWhitelist: {
+      fqdns: fqdns
+      ipAddresses: ipAddresses
+    }
+    transportSecurity: (!empty(keyVaultSecretId) || !empty(certificateName)) ? {
+      certificateAuthority: {
+        keyVaultSecretId: !empty(keyVaultSecretId) ? keyVaultSecretId : null
+        name: !empty(certificateName) ? certificateName : null
+      }
+    } : null
+  }
+}
+
+module ruleCollectionGroups_resource 'ruleCollectionGroups/deploy.bicep' = [for (ruleCollectionGroup, index) in ruleCollectionGroups: {
+  name: '${uniqueString(deployment().name, location)}-ruleCollectionGroup-${index}'
```

```suggestion
  name: '${uniqueString(deployment().name, location)}-firewallPolicy_ruleCollectionGroup-${index}'
```

Update as per [documentation](https://github.com/Azure/ResourceModules/wiki/ModulesDesign#deployment-names).

---

done